source/appModuleHandler.py

```diff
@@ -344,19 +344,55 @@ def __init__(self,processID,appName=None):
 
 	def _setProductInfo(self):
 		"""Set productName and productVersion attributes.
+		There are at least two ways of obtaining product info for an app:
+		* Package info for hosted apps
+		* File version info for other apps and for some hosted apps
 		"""
 		# Sometimes (I.E. when NVDA starts) handle is 0, so stop if it is the case
 		if not self.processHandle:
 			raise RuntimeError("processHandle is 0")
-		# Create the buffer to get the executable name
-		exeFileName = ctypes.create_unicode_buffer(ctypes.wintypes.MAX_PATH)
-		length = ctypes.wintypes.DWORD(ctypes.wintypes.MAX_PATH)
-		if not ctypes.windll.Kernel32.QueryFullProcessImageNameW(self.processHandle, 0, exeFileName, ctypes.byref(length)):
-			raise ctypes.WinError()
-		fileName = exeFileName.value
-		fileinfo = getFileVersionInfo(fileName, "ProductName", "ProductVersion")
-		self.productName = fileinfo["ProductName"]
-		self.productVersion = fileinfo["ProductVersion"]
+		# Use an internal function for obtaining file name and version for the executable.
+		# This is needed in case immersive app package returns an error,
+		# dealing with a native app, or a converted desktop app.
+
+		def _getExecutableFileInfo():

```

Comment:
I think I would prefer this to be a private appModule method instead of an internal function to _setProductInfo.